### ISSUE TITLE
[1.29.0]

### ISSUE THREAD

**Issue by firojcmulani** [mobile-dev-inc/maestro#1193]
**Describe the bug**
parallel Execution is not working on Maestor Cloud
**To Reproduce**
Steps to reproduce the behavior, e.g.:

1. Have 2 yaml files under .maestro/TM folder   e.g. Demo_English.yaml and Demo_Hindi.yaml
2. Have .apk file under ./maestro folder
3. Open wsl command line tool and go to .maestro folder and then run maestro cloud TM_2.0.80.apk \TM - Observe that the first Execution happens for Demo_English.yaml and then for Demo_Hindi.yaml ie. sequential execution instead of Parallel execution

**Expected behavior**
Parallel execution should work.
**Screenshots**

**Environment information (please complete the following information):**

- Maestro version [e.g. v1.2.3]
- Platform: [e.g. iOS 16.4.1, Android 11.0, Chrome]
- Framework: [Native iOS/Android, Flutter, React Native, etc]
- Device model and OS version: [e.g. iPhone6]
- Simulator, emulator or physical device
- Host [e.g. Mac M1, Windows, Linux, etc]

**Workspace (if applicable)**
Please share your workspace with Flow files and application binary if you are able to do so as that greatly helps in reproduction efforts.

**Bugreport output (if applicable)**
The zip files created when `maestro bugreport` can be uploaded here as well if applicable.

**Additional context**

**Comment by firojcmulani** [comment]
Demo_English.yaml:
appId: "com.tejimandi.android.beta"
---
- launchApp: 
    appId: "com.tejimandi.android.beta"
    clearState: true   
- tapOn: "English"
- tapOn: "Next"


Demo_Hindi.yaml:
appId: "com.tejimandi.android.beta"
---
- launchApp: 
    appId: "com.tejimandi.android.beta"
    clearState: true   
- tapOn: "हिंदी"
- tapOn: "Next"

**Comment by axelniklasson** [comment]
Hi there @firojcmulani! Parallel execution is the default mode in Cloud, so it sounds weird that it is not working for you. Are you referring to the fact that the output is processed sequentially in the CLI when you run `maestro cloud` by any chance?

<img width="375" alt="image" src="https://github.com/mobile-dev-inc/maestro/assets/6113675/97e1ac55-f439-4cfa-8e3b-89b0ea0ec8c7">

**Comment by firojcmulani** [comment]
@axelniklasson Right, I am referring to the fact that the output is processed sequentially in the CLI and also maestro cloud account.

**Comment by axelniklasson** [comment]
@firojcmulani That doesn't sound correct -- if you don't specify execution order, Flows will be run in parallel and not in sequence. I just verified that this is the behaviour. However, note that depending on the current load on the system the Flows might have to wait in the queue to be picked up for a while and since your Flows are very short the first one might be finished before the second one is picked up. 

I just ran a test upload on my account and Flows are being executed in parallel, so I am unable to reproduce this issue. Could you share your `.maestro` folder and a link to your console with me?

**Comment by firojcmulani** [comment]
@axelniklasson We can close this issue as it's an invalid, I thought that the output comes sequentially, so it's not executing parallelly.
